src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlRetry/SqlRetryService.cs

```diff
@@ -334,5 +346,31 @@ public async Task ExecuteSql<TLogger>(SqlCommand sqlCommand, Func<SqlCommand, Ca
             List<TResult> result = await ExecuteSqlDataReader(sqlCommand, readerToResult, logger, logMessage, false, cancellationToken);
             return result.Count > 0 ? result[0] : null;
         }
+
+        private async Task TryLogToDatabase(string process, string status, string text, DateTime? startDate, CancellationToken cancellationToken)
+        {
+            try
+            {
+                using var cmd = new SqlCommand() { CommandType = CommandType.StoredProcedure, CommandText = "dbo.LogEvent" };
+                cmd.Parameters.AddWithValue("@Process", process);
+                cmd.Parameters.AddWithValue("@Status", status);
+                cmd.Parameters.AddWithValue("@Text", text);
+                if (startDate.HasValue)
+                {
+                    cmd.Parameters.AddWithValue("@Start", startDate.Value);
+                }
+
+                using var conn = await _sqlConnectionBuilder.GetSqlConnectionAsync(initialCatalog: null, cancellationToken: cancellationToken).ConfigureAwait(false);
+                conn.RetryLogicProvider = null;
+                await conn.OpenAsync(cancellationToken);
+                cmd.Connection = conn;
+
+                await cmd.ExecuteNonQueryAsync(cancellationToken);
+            }
+            catch
+            {
+                // do nothing;
```

Comment:
Does it make sense to, in case of failure, log the error using our regular logging framework?

Reply:
I don't think so. When we do Log.Whatever we do our best effort, and do not try to log if log itself fails.